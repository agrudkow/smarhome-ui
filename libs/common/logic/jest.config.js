module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
