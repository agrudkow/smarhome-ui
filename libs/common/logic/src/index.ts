export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
