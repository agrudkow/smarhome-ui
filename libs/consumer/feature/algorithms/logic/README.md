# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
