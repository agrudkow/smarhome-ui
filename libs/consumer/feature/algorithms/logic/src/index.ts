export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
