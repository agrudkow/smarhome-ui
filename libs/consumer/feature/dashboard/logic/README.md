# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
