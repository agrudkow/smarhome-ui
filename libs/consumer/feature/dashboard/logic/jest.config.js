module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
