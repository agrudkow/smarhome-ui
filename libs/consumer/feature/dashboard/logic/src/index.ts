export * from './lib/use-billings';
