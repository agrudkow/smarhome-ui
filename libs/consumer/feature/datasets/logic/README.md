# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
