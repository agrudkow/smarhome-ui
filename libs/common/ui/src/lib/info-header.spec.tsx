import React from 'react';
import { customRender } from '@smarthome/common/logic';
import InfoHeader from './info-header';

describe(' InfoHeader', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <InfoHeader headerText="test" infoMessageText="test" />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <InfoHeader headerText="test" infoMessageText="test" />
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });
});
