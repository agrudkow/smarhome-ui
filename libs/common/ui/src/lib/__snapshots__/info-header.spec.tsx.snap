// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` InfoHeader should match snapshot 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  color: #444;
  font-size: 20px;
  font-weight: bold;
  line-height: 20px;
}

.c3 {
  margin: 0 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  padding: 5px;
  width: calc(100% - 14px);
  height: 0;
  margin: 0;
  visibility: hidden;
  overflow: hidden;
  -webkit-transition: height 0.3s ease-in-out,visibility 0.3s ease-in-out, margin 0.3s ease-in-out;
  transition: height 0.3s ease-in-out,visibility 0.3s ease-in-out, margin 0.3s ease-in-out;
  border: 2px dashed #1BBDCE;
  border-radius: 6px;
  text-align: justify;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 0.8;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:1024px) {
  .c2 {
    font-size: 30px;
    -webkit-letter-spacing: -0.3px;
    -moz-letter-spacing: -0.3px;
    -ms-letter-spacing: -0.3px;
    letter-spacing: -0.3px;
    line-height: 30px;
  }
}

@media (min-width:769px) {
  .c4 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c4 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (max-width:480px) {
  .c4 {
    margin: 0;
  }
}

@media (max-width:480px) {
  .c1 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        test
      </h3>
      <div
        class="c3"
      >
        <button
          aria-label="info"
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="info-button-icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        test
      </div>
    </div>
  </div>
</div>
`;
